pytorch_lightning/trainer/trainer.py

```diff
@@ -1248,10 +1248,10 @@ def _log_hyperparams(self) -> None:
         # log hyper-parameters
         hparams_initial = None
 
-        if self.logger is not None:
-            # save exp to get started (this is where the first experiment logs are written)
-            datamodule_log_hyperparams = self.datamodule._log_hyperparams if self.datamodule is not None else False
+        # save exp to get started (this is where the first experiment logs are written)
+        datamodule_log_hyperparams = self.datamodule._log_hyperparams if self.datamodule is not None else False
 
+        if self.loggers:
```

Comment:
why not move this check to the top of the method and return early if there are no loggers?

Reply:
good suggestion!